packages/analytics/amplify_analytics_pinpoint_dart/lib/src/analytics_plugin_impl.dart

```diff
@@ -153,11 +153,22 @@ class AmplifyAnalyticsPinpointDart extends AnalyticsPluginInterface {
       },
     );
 
-    _autoEventSubmitter = StoppableTimer(
-      duration: const Duration(seconds: 10),
-      callback: flushEvents,
-      onError: (e) => _logger.warn('Exception in events auto flush', e),
-    );
+    final autoFlushEventsInterval = pinpointConfig.autoFlushEventsInterval;
+
+    if (autoFlushEventsInterval.isNegative) {
```

Comment:
What about 0?

Reply:
0 is implicitly handled - 0 interval means no autoflush.

Current conditions are:
if negative throw config error
if duration > 0 do autoflush
implicit case --> no autoflush, duration was 0